game-core/src/main/java/games/strategy/triplea/delegate/BattleDelegate.java

```diff
@@ -819,28 +814,26 @@ private void doScrambling() {
         continue;
       }
       boolean scrambledHere = false;
-      for (final var scramblers : scramblersByTerritoryPlayer.get(terrPlayer)) {
-        // verify that we didn't already scramble any of these units
-        final Iterator<Territory> territoryIter = scramblers.keySet().iterator();
-        while (territoryIter.hasNext()) {
-          final Territory t = territoryIter.next();
-          scramblers.get(t).getSecond().retainAll(t.getUnitCollection());
-          if (scramblers.get(t).getSecond().isEmpty()) {
-            territoryIter.remove();
```

Comment:
This logic was removing an entry from the scramblers map when it no longer had units.
This was fine, but really it's unnecessary complex. I replaced it with logic to just count
the remaining units instead sand only proceed if there are any left, which is equivalent.

Reply:
Actually, looks like this is causing a behaviour changes - although very subtle.

This can result in empty map entries being passed to the player via scrambleUnitsQuery(), if there is at least one entry that's not empty. This actually causes the UI for scrambling to show a box with no units rather than not show a box. Since I didn't intend a behavior change here, let me restore the previous logic.

Reply:
Restored the logic with a slightly better construct.